app/queues/common/expiration-request.server.ts

```diff
@@ -0,0 +1,101 @@
+import { Worker, Queue, UnrecoverableError } from 'bullmq';
+
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { prisma } from '~/db.server';
+
+import { addNotification } from '../notifications/notifications.server';
+import { deleteDnsRequest } from '../dns/delete-record-flow.server';
+import { RecordType } from '../notifications/expiration-notification.server';
+
+import type { ExpirationStatusPayload as ExpirationRequestPayload } from '../notifications/expiration-notification.server';
+
+const { EXPIRATION_REPEAT_FREQUENCY_MS, JOB_REMOVAL_FREQUENCY_MS } = process.env;
+const JOB_REMOVAL_INTERVAL = 7 * 24 * 60 * 60 * 1000;
+declare global {
+  var __expiration_request_init__: boolean;
+}
+// queue name
+const expirationRequestQueueName = 'expiration-request';
+
+// query to fetch expired records from database based on type
+const getExpiredRecords = (type: RecordType) => {
```

Comment:
I'd suggest to create two separate functions, one for records, one for certificates. I'd probably move this functionality to the records and certificates model files too.

Reply:
Agree, this can move up a level in our API.